docs/config-s3gw.md

```diff
@@ -0,0 +1,322 @@
+at the various Ingresses and ClusterIP resources.
+Note that the connection between the Ingress and the s3gw's ClusterIP
+within the cluster will not be TLS protected.
+
+```yaml
+tls:
+  publicDomain:
+    crt: PUBLIC_DOMAIN_CERTIFICATE_HERE
+    key: PUBLIC_DOMAIN_CERTIFICATE_KEY_HERE
+  privateDomain:
+    crt: PRIVATE_DOMAIN_CERTIFICATE_HERE
+    key: PRIVATE_DOMAIN_CERTIFICATE_KEY_HERE
+  ui:
+    publicDomain:
+      crt: CERTIFICATE_FOR_UI
+      key: CERTIFICATE_KEY_FOR_UI
+```
+
+The certificates must be provided as base64 encoded PEM in one long
+string without line breaks. You can create them from a PEM file.
+
+## s3gw user interface
+
+`ui.enabled` is by default set to `true`, but can be set to `false` to use
+the s3gw standalone.
+
+`ui.serviceName` and `ui.publicDomain` are the hostname and domain name
+which the ingress will listed on for access to the UI. If the UI is set
+as such:
+
+```yaml
+  serviceName: object-browser
+  publicDomain: example.com
+```
+
+The UI can be accessed under `http(s)://object-browser.example.com`.
+
+## S3 service
+
+`useExistingSecret` can be used to tell the chart that you want to provide a
+secret where the access credentials for the first account can be found.
+`defaultUserCredentialsSecret` can be used to tell helm which secret that will be.
+
+If `accessKey` and `secretKey` are left empty, credentials will be generated
+automatically.
+
+### Service name
+
+There are two possible ways to access the s3gw: from inside the Kubernetes
+cluster and from the outside. For both, the s3gw must be configured with the
+correct service and domain name. Use the `publicDomain` and the
+`ui.publicDomain` setting to configure the domain under which the s3gw and the
+UI respectively be available to the outside of the Kubernetes cluster. Use the
+`privateDomain` setting to set the cluster's internal domain and make the s3gw
+available inside the cluster to other deployments.
+
+```yaml
+serviceName: s3gw
+publicDomain: "fe.127.0.0.1.omg.howdoi.website"
+privateDomain: "s3gw-namespace.svc.cluster.local"
+ui:
+  serviceName: s3gw-ui
+  publicDomain: "be.127.0.0.1.omg.howdoi.website"
+```
+
+## Storage
+
+The s3gw is best deployed on top of a [Longhorn](https://longhorn.io) volume.
+If you have Longhorn installed in your cluster, all appropriate resources are
+automatically deployed for you.
+
+The size of the volume can be controlled with `storageSize`:
+
+```yaml
+storageSize: 10Gi
+```
+
+If you want to reuse an existing storage class or otherwise need more control
+over storage settings, set `storageClass.create` to `false` and
+`storageClass.name` to the name of your preferred storage class.
+
+```yaml
+storageClass:
+  name: my-custom-storageclass
+  create: false
+```
+
+### Local storage
+
+You can use the `storageClass.local` and `storageClass.localPath` variables to
+set up a node-local volume for testing if you don not have Longhorn. This is an
+experimental feature for development use only.
```

Comment:
Maybe the "experimental feature"-sentence could be put as a ``**NOTE::**``?

Reply:
Also there's a typo here: `... don not have Longhorn...`

Reply:
don is good